cli/src/main/java/com/box/l10n/mojito/cli/command/checks/SpellCliChecker.java

```diff
@@ -0,0 +1,228 @@
+        while(index != -1 && stringWithoutPlaceholders.contains("}")) {
+            int associatedClosingBraceIndex = getEndOfPlaceholderIndex(stringWithoutPlaceholders, index);
+            String tmp = stringWithoutPlaceholders.substring(index, associatedClosingBraceIndex + 1 < stringWithoutPlaceholders.length()? associatedClosingBraceIndex + 1 : associatedClosingBraceIndex).replaceAll("\\{", "\\\\{").replaceAll("\\}", "\\\\}");
+            stringWithoutPlaceholders = stringWithoutPlaceholders.replaceAll(tmp, "");
+            index = stringWithoutPlaceholders.indexOf("{");
+        }
+        return stringWithoutPlaceholders;
+    }
+
+    private int getEndOfPlaceholderIndex(String str, int startIndex){
+        ArrayDeque<Character> stack = new ArrayDeque<>();
+        int indexCount = startIndex;
+        for (Character c : str.substring(startIndex).toCharArray()) {
+            if (c.equals('{')) {
+                stack.push(c);
+                indexCount++;
+                continue;
+            } else if (c.equals('}')) {
+                if(stack.isEmpty()){
+                    throw new CommandException("Invalid number of opening brackets in string.");
+                }
+                stack.pop();
+                if(stack.isEmpty()) {
+                    return indexCount;
+                }
+            }
+            indexCount++;
+        }
+        if(!stack.isEmpty()) {
+            throw new CommandException("Invalid number of closing brackets in string.");
+        }
+        return -1;
+    }
+
+    private void loadAdditionalWordsToDictionary(String additionalWordsFilePath) {
+
+        if (additionalWordsFilePath != null && !additionalWordsFilePath.isEmpty()) {
+            try {
+                Files.readAllLines(Paths.get(additionalWordsFilePath)).stream().forEach(word -> {
+                    String w = word.trim();
+                    logger.debug("Adding {} to dictionary", w);
+                    hunspell.add(w);
+                });
+            } catch (IOException e) {
+                logger.error("Error adding additional words to dictionary: {}", e);
+                throw new CommandException("Error adding additional words to dictionary: " + e.getMessage());
+            }
+        }
+    }
+
+    private Map<String, List<String>> spellCheckSourceString(List<String> words) {
+        Map<String, List<String>> failureMap = new HashMap<>();
+        Set<String> checked = new HashSet<>();
+        for (String word : words) {
+            if (!checked.contains(word) && !hunspell.spell(word)) {
+                List<String> suggestions = hunspell.suggest(word);
+                logger.debug("{} is spelt incorrectly. Suggested correct spellings are {}", word, suggestions);
+                failureMap.put(word, suggestions);
+                checked.add(word);
+            }
+        }
+        return failureMap;
+    }
+
+    private List<String> getSourceStringsFromDiff() {
```

Comment:
do you foresee this to be used by more checkers? should it be in the abstract class?